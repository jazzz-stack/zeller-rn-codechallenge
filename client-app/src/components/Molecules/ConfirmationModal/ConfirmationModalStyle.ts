import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  overlay: {
    flex: 1,
    backgroundColor: 'rgba(0, 0, 0, 0.5)',
    justifyContent: 'center',
    alignItems: 'center',
    paddingHorizontal: 20,
  },
  modalContainer: {
    backgroundColor: '#FFFFFF',
    borderRadius: 16,
    paddingVertical: 24,
    paddingHorizontal: 20,
    minWidth: 300,
    maxWidth: 340,
    alignItems: 'center',
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 8,
    },
    shadowOpacity: 0.25,
    shadowRadius: 16,
    elevation: 16,
  },
  iconContainer: {
    marginBottom: 16,
    padding: 12,
    backgroundColor: '#FFF5F5',
    borderRadius: 50,
  },
  title: {
    fontSize: 20,
    fontWeight: '600',
    color: '#1D1D1F',
    textAlign: 'center',
    marginBottom: 8,
  },
  message: {
    fontSize: 16,
    color: '#6D6D80',
    textAlign: 'center',
    lineHeight: 22,
    marginBottom: 24,
  },
  buttonContainer: {
    flexDirection: 'row',
    gap: 12,
    width: '100%',
  },
  button: {
    flex: 1,
    paddingVertical: 12,
    paddingHorizontal: 16,
    borderRadius: 8,
    alignItems: 'center',
    justifyContent: 'center',
    minHeight: 44,
  },
  cancelButton: {
    backgroundColor: '#F2F2F7',
    borderWidth: 1,
    borderColor: '#E5E5EA',
  },
  confirmButton: {
    backgroundColor: '#FF3B30',
  },
  singleButton: {
    flex: 1,
  },
  disabledButton: {
    backgroundColor: '#FFB3B3',
    opacity: 0.7,
  },
  cancelButtonText: {
    fontSize: 16,
    fontWeight: '500',
    color: '#1D1D1F',
  },
  confirmButtonText: {
    fontSize: 16,
    fontWeight: '500',
    color: '#FFFFFF',
  },
  loadingContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
