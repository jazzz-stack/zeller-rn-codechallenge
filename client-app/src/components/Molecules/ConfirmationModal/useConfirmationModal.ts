import React from "react";
import { Animated } from "react-native";

export const useConfirmationModal = ({
  visible,
  title,
  message,
  customerName,
  confirmText,
  type,
}) => {
  const scaleValue = React.useRef(new Animated.Value(0)).current;

  React.useEffect(() => {
    if (visible) {
          Animated.spring(scaleValue, {
            toValue: 1,
            useNativeDriver: true,
            tension: 100,
            friction: 8,
          }).start();
        } else {
          Animated.timing(scaleValue, {
            toValue: 0,
            duration: 200,
            useNativeDriver: true,
          }).start();
        }
      }, [visible, scaleValue]);
    
    

  const getIconConfig = () => {
    switch (type) {
      case 'success':
        return {name: 'check-circle', color: '#4CAF50'};
      case 'error':
        return {name: 'error', color: '#FF3B30'};
      case 'warning':
      case 'delete':
        return {name: 'warning', color: '#f8b50aff'};
      default:
        return {name: 'info', color: '#007AFF'};
    }
  };

  const getButtonConfig = () => {
    switch (type) {
      case 'success':
        return {backgroundColor: '#4CAF50'};
      case 'error':
        return {backgroundColor: '#FF3B30'};
      case 'warning':
        return {backgroundColor: '#f8b50aff'};
      case 'delete':
        return {backgroundColor: '#FF3B30'};
      default:
        return {backgroundColor: '#007AFF'};
    }
  };

  const getDefaultTitle = () => {
    switch (type) {
      case 'success':
        return title || 'Success';
      case 'error':
        return title || 'Error';
      case 'warning':
        return title || 'Warning';
      case 'delete':
        return title || 'Delete Customer';
      default:
        return title || 'Information';
    }
  };

  const getDefaultConfirmText = () => {
    switch (type) {
      case 'delete':
        return confirmText === 'OK' ? 'Delete' : confirmText;
      default:
        return confirmText;
    }
  };

  const defaultMessage =
    customerName && type === 'delete'
      ? `Are you sure you want to delete ${customerName}?`
      : message;

  const displayMessage = defaultMessage;
  const iconConfig = getIconConfig();
  const buttonConfig = getButtonConfig();
  const displayTitle = getDefaultTitle();
  const displayConfirmText = getDefaultConfirmText();

  return {
    iconConfig,
    buttonConfig,
    displayTitle,
    displayMessage,
    displayConfirmText,
    scaleValue,
  };
}