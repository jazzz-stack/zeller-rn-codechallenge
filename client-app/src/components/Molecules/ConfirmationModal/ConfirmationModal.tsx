import React, {use} from 'react';
import {
  Modal,
  View,
  Text,
  TouchableOpacity,
  TouchableWithoutFeedback,
  Animated,
} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import {styles} from './ConfirmationModalStyle';
import {useConfirmationModal} from './useConfirmationModal';

interface ConfirmationModalProps {
  visible: boolean;
  title?: string;
  message: string;
  customerName?: string; // Keep for backward compatibility
  onConfirm: () => void;
  onCancel?: () => void;
  confirmText?: string;
  cancelText?: string;
  isLoading?: boolean;
  type?: 'success' | 'error' | 'warning' | 'info' | 'delete';
  showCancel?: boolean;
}

const ConfirmationModal: React.FC<ConfirmationModalProps> = ({
  visible,
  title,
  message,
  customerName,
  onConfirm,
  onCancel,
  confirmText = 'OK',
  cancelText = 'Cancel',
  isLoading = false,
  type = 'info',
  showCancel = true,
}) => {
  const {
    iconConfig,
    buttonConfig,
    displayTitle,
    displayMessage,
    displayConfirmText,
    scaleValue,
  } = useConfirmationModal({
    visible,
    title,
    message,
    customerName,
    confirmText,
    type,
  });
  
  return (
    <Modal
      visible={visible}
      transparent
      animationType="fade"
      statusBarTranslucent
      onRequestClose={onCancel || (() => {})}>
      <TouchableWithoutFeedback onPress={onCancel || (() => {})}>
        <View style={styles.overlay}>
          <TouchableWithoutFeedback>
            <Animated.View
              style={[
                styles.modalContainer,
                {
                  transform: [{scale: scaleValue}],
                },
              ]}>
              <View style={styles.iconContainer}>
                <Icon
                  name={iconConfig.name}
                  size={48}
                  color={iconConfig.color}
                />
              </View>

              <Text style={styles.title}>{displayTitle}</Text>
              <Text style={styles.message}>{displayMessage}</Text>

              <View style={styles.buttonContainer}>
                {showCancel && onCancel && (
                  <TouchableOpacity
                    style={[styles.button, styles.cancelButton]}
                    onPress={onCancel}
                    disabled={isLoading}>
                    <Text style={styles.cancelButtonText}>{cancelText}</Text>
                  </TouchableOpacity>
                )}

                <TouchableOpacity
                  style={[
                    styles.button,
                    styles.confirmButton,
                    {backgroundColor: buttonConfig.backgroundColor},
                    !showCancel && styles.singleButton,
                    isLoading && styles.disabledButton,
                  ]}
                  onPress={onConfirm}
                  disabled={isLoading}>
                  {isLoading ? (
                    <View style={styles.loadingContainer}>
                      <Text style={styles.confirmButtonText}>
                        {type === 'delete' ? 'Deleting...' : 'Please wait...'}
                      </Text>
                    </View>
                  ) : (
                    <Text style={styles.confirmButtonText}>
                      {displayConfirmText}
                    </Text>
                  )}
                </TouchableOpacity>
              </View>
            </Animated.View>
          </TouchableWithoutFeedback>
        </View>
      </TouchableWithoutFeedback>
    </Modal>
  );
};

export default ConfirmationModal;
