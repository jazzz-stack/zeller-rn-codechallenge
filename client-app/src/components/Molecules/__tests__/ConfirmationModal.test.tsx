import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react-native';
import { Modal, TouchableWithoutFeedback } from 'react-native';
import ConfirmationModal from '../ConfirmationModal/ConfirmationModal';

// Mock react-native-vector-icons
jest.mock('react-native-vector-icons/MaterialIcons', () => 'Icon');

describe('ConfirmationModal', () => {
  const defaultProps = {
    visible: true,
    message: 'Are you sure you want to delete this customer?',
    onConfirm: jest.fn(),
    onCancel: jest.fn(),
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders with default props', () => {
    render(<ConfirmationModal {...defaultProps} />);
    
    expect(screen.getByText('Information')).toBeTruthy(); // Default title for type 'info'
    expect(screen.getByText('Are you sure you want to delete this customer?')).toBeTruthy();
    expect(screen.getByText('Cancel')).toBeTruthy();
    expect(screen.getByText('OK')).toBeTruthy(); // Default confirm text
  });

  it('renders with customer name', () => {
    render(<ConfirmationModal {...defaultProps} type="delete" customerName="John Doe" />);
    
    expect(screen.getByText('Delete Customer')).toBeTruthy();
    // Should show the generated message with customer name
    expect(screen.getByText('Are you sure you want to delete John Doe?')).toBeTruthy();
  });

  it('renders with custom title and message', () => {
    render(
      <ConfirmationModal 
        {...defaultProps} 
        title="Custom Title"
        message="Custom message"
      />
    );
    
    expect(screen.getByText('Custom Title')).toBeTruthy();
    expect(screen.getByText('Custom message')).toBeTruthy();
  });

  it('renders with custom button text', () => {
    render(
      <ConfirmationModal 
        {...defaultProps} 
        confirmText="Remove"
        cancelText="Keep"
      />
    );
    
    expect(screen.getByText('Keep')).toBeTruthy();
    expect(screen.getByText('Remove')).toBeTruthy();
  });

  it('calls onCancel when cancel button is pressed', () => {
    render(<ConfirmationModal {...defaultProps} />);
    
    fireEvent.press(screen.getByText('Cancel'));
    expect(defaultProps.onCancel).toHaveBeenCalledTimes(1);
  });

  it('calls onConfirm when confirm button is pressed', () => {
    render(<ConfirmationModal {...defaultProps} />);
    
    fireEvent.press(screen.getByText('OK')); // Updated to use default confirm text
    expect(defaultProps.onConfirm).toHaveBeenCalledTimes(1);
  });

  it('shows loading state when isLoading is true', () => {
    render(<ConfirmationModal {...defaultProps} type="delete" isLoading={true} />);
    
    expect(screen.getByText('Deleting...')).toBeTruthy();
  });

  it('shows generic loading state for non-delete types', () => {
    render(<ConfirmationModal {...defaultProps} isLoading={true} />);
    
    expect(screen.getByText('Please wait...')).toBeTruthy();
  });

  it('disables buttons when isLoading is true', () => {
    render(<ConfirmationModal {...defaultProps} type="delete" isLoading={true} />);
    
    // Check that loading text is shown and buttons exist (disabled state is handled by TouchableOpacity disabled prop)
    expect(screen.getByText('Deleting...')).toBeTruthy();
    expect(screen.getByText('Cancel')).toBeTruthy();
  });

  it('does not render when visible is false', () => {
    render(<ConfirmationModal {...defaultProps} visible={false} />);
    
    expect(screen.queryByText('Information')).toBeNull(); // Updated to use default title
  });

  it('renders different types correctly', () => {
    const { rerender } = render(<ConfirmationModal {...defaultProps} type="success" />);
    expect(screen.getByText('Success')).toBeTruthy();
    
    rerender(<ConfirmationModal {...defaultProps} type="error" />);
    expect(screen.getByText('Error')).toBeTruthy();
    
    rerender(<ConfirmationModal {...defaultProps} type="warning" />);
    expect(screen.getByText('Warning')).toBeTruthy();
    
    rerender(<ConfirmationModal {...defaultProps} type="delete" />);
    expect(screen.getByText('Delete Customer')).toBeTruthy();
  });

  it('hides cancel button when showCancel is false', () => {
    render(<ConfirmationModal {...defaultProps} showCancel={false} />);
    
    expect(screen.queryByText('Cancel')).toBeNull();
    expect(screen.getByText('OK')).toBeTruthy();
  });

  // Snapshot Tests
  it('matches snapshot with default props', () => {
    const component = render(<ConfirmationModal {...defaultProps} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('matches snapshot with loading state', () => {
    const component = render(<ConfirmationModal {...defaultProps} isLoading={true} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('matches snapshot with customer name', () => {
    const component = render(<ConfirmationModal {...defaultProps} customerName="John Doe" />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('handles missing onCancel prop gracefully', () => {
    const propsWithoutCancel = {
      visible: true,
      message: 'Test message',
      onConfirm: jest.fn(),
      // No onCancel prop
    };
    
    render(<ConfirmationModal {...propsWithoutCancel} />);
    expect(screen.getByText('Test message')).toBeTruthy();
  });

  it('renders without cancel button when showCancel is false and onCancel is not provided', () => {
    const propsWithoutCancel = {
      visible: true,
      message: 'Test message',
      onConfirm: jest.fn(),
      showCancel: false,
    };
    
    render(<ConfirmationModal {...propsWithoutCancel} />);
    expect(screen.queryByText('Cancel')).toBeNull();
    expect(screen.getByText('OK')).toBeTruthy();
  });

  it('renders with showCancel true but no onCancel provided', () => {
    const propsWithoutCancel = {
      visible: true,
      message: 'Test message',
      onConfirm: jest.fn(),
      showCancel: true,
      // No onCancel prop
    };
    
    render(<ConfirmationModal {...propsWithoutCancel} />);
    expect(screen.queryByText('Cancel')).toBeNull(); // Should not show cancel button if no onCancel
    expect(screen.getByText('OK')).toBeTruthy();
  });

  it('calls fallback onRequestClose when onCancel is not provided', () => {
    const propsWithoutCancel = {
      visible: true,
      message: 'Test message',
      onConfirm: jest.fn(),
      // No onCancel prop
    };
    const { UNSAFE_getByType } = render(<ConfirmationModal {...propsWithoutCancel} />);
    // Find the Modal and call onRequestClose
    const modal = UNSAFE_getByType(Modal);
    expect(() => modal.props.onRequestClose()).not.toThrow();
  });

  it('calls fallback onPress for backdrop when onCancel is not provided', () => {
    const propsWithoutCancel = {
      visible: true,
      message: 'Test message',
      onConfirm: jest.fn(),
      // No onCancel prop
    };
    const { UNSAFE_getAllByType } = render(<ConfirmationModal {...propsWithoutCancel} />);
    // The first TouchableWithoutFeedback is the backdrop
    const backdrop = UNSAFE_getAllByType(TouchableWithoutFeedback)[0];
    expect(() => backdrop.props.onPress()).not.toThrow();
  });
});
