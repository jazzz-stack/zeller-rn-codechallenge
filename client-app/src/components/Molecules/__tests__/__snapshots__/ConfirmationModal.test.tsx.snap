// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationModal matches snapshot with customer name 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  statusBarTranslucent={true}
  transparent={true}
  visible={true}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.5)",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 20,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 16,
          "elevation": 16,
          "maxWidth": 340,
          "minWidth": 300,
          "paddingHorizontal": 20,
          "paddingVertical": 24,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 8,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 16,
          "transform": [
            {
              "scale": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#FFF5F5",
            "borderRadius": 50,
            "marginBottom": 16,
            "padding": 12,
          }
        }
      >
        <Icon
          color="#007AFF"
          name="info"
          size={48}
        />
      </View>
      <Text
        style={
          {
            "color": "#1D1D1F",
            "fontSize": 20,
            "fontWeight": "600",
            "marginBottom": 8,
            "textAlign": "center",
          }
        }
      >
        Information
      </Text>
      <Text
        style={
          {
            "color": "#6D6D80",
            "fontSize": 16,
            "lineHeight": 22,
            "marginBottom": 24,
            "textAlign": "center",
          }
        }
      >
        Are you sure you want to delete this customer?
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "gap": 12,
            "width": "100%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#F2F2F7",
              "borderColor": "#E5E5EA",
              "borderRadius": 8,
              "borderWidth": 1,
              "flex": 1,
              "justifyContent": "center",
              "minHeight": 44,
              "opacity": 1,
              "paddingHorizontal": 16,
              "paddingVertical": 12,
            }
          }
        >
          <Text
            style={
              {
                "color": "#1D1D1F",
                "fontSize": 16,
                "fontWeight": "500",
              }
            }
          >
            Cancel
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#007AFF",
              "borderRadius": 8,
              "flex": 1,
              "justifyContent": "center",
              "minHeight": 44,
              "opacity": 1,
              "paddingHorizontal": 16,
              "paddingVertical": 12,
            }
          }
        >
          <Text
            style={
              {
                "color": "#FFFFFF",
                "fontSize": 16,
                "fontWeight": "500",
              }
            }
          >
            OK
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`ConfirmationModal matches snapshot with default props 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  statusBarTranslucent={true}
  transparent={true}
  visible={true}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.5)",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 20,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 16,
          "elevation": 16,
          "maxWidth": 340,
          "minWidth": 300,
          "paddingHorizontal": 20,
          "paddingVertical": 24,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 8,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 16,
          "transform": [
            {
              "scale": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#FFF5F5",
            "borderRadius": 50,
            "marginBottom": 16,
            "padding": 12,
          }
        }
      >
        <Icon
          color="#007AFF"
          name="info"
          size={48}
        />
      </View>
      <Text
        style={
          {
            "color": "#1D1D1F",
            "fontSize": 20,
            "fontWeight": "600",
            "marginBottom": 8,
            "textAlign": "center",
          }
        }
      >
        Information
      </Text>
      <Text
        style={
          {
            "color": "#6D6D80",
            "fontSize": 16,
            "lineHeight": 22,
            "marginBottom": 24,
            "textAlign": "center",
          }
        }
      >
        Are you sure you want to delete this customer?
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "gap": 12,
            "width": "100%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#F2F2F7",
              "borderColor": "#E5E5EA",
              "borderRadius": 8,
              "borderWidth": 1,
              "flex": 1,
              "justifyContent": "center",
              "minHeight": 44,
              "opacity": 1,
              "paddingHorizontal": 16,
              "paddingVertical": 12,
            }
          }
        >
          <Text
            style={
              {
                "color": "#1D1D1F",
                "fontSize": 16,
                "fontWeight": "500",
              }
            }
          >
            Cancel
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#007AFF",
              "borderRadius": 8,
              "flex": 1,
              "justifyContent": "center",
              "minHeight": 44,
              "opacity": 1,
              "paddingHorizontal": 16,
              "paddingVertical": 12,
            }
          }
        >
          <Text
            style={
              {
                "color": "#FFFFFF",
                "fontSize": 16,
                "fontWeight": "500",
              }
            }
          >
            OK
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`ConfirmationModal matches snapshot with loading state 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  statusBarTranslucent={true}
  transparent={true}
  visible={true}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.5)",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 20,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 16,
          "elevation": 16,
          "maxWidth": 340,
          "minWidth": 300,
          "paddingHorizontal": 20,
          "paddingVertical": 24,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 8,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 16,
          "transform": [
            {
              "scale": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#FFF5F5",
            "borderRadius": 50,
            "marginBottom": 16,
            "padding": 12,
          }
        }
      >
        <Icon
          color="#007AFF"
          name="info"
          size={48}
        />
      </View>
      <Text
        style={
          {
            "color": "#1D1D1F",
            "fontSize": 20,
            "fontWeight": "600",
            "marginBottom": 8,
            "textAlign": "center",
          }
        }
      >
        Information
      </Text>
      <Text
        style={
          {
            "color": "#6D6D80",
            "fontSize": 16,
            "lineHeight": 22,
            "marginBottom": 24,
            "textAlign": "center",
          }
        }
      >
        Are you sure you want to delete this customer?
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "gap": 12,
            "width": "100%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#F2F2F7",
              "borderColor": "#E5E5EA",
              "borderRadius": 8,
              "borderWidth": 1,
              "flex": 1,
              "justifyContent": "center",
              "minHeight": 44,
              "opacity": 1,
              "paddingHorizontal": 16,
              "paddingVertical": 12,
            }
          }
        >
          <Text
            style={
              {
                "color": "#1D1D1F",
                "fontSize": 16,
                "fontWeight": "500",
              }
            }
          >
            Cancel
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#FFB3B3",
              "borderRadius": 8,
              "flex": 1,
              "justifyContent": "center",
              "minHeight": 44,
              "opacity": 0.7,
              "paddingHorizontal": 16,
              "paddingVertical": 12,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                  "fontWeight": "500",
                }
              }
            >
              Please wait...
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
